### PR TITLE
refactor(process_tracker): move process_tracker to `scheduler` from `diesel_models`

### LINKED ISSUE #4921: Move Process Tracker files from `diesel_models` crate to `scheduler` crate
Currently, `diesel_models` crate has `process_tracker` (`src/process_tracker.rs`, `src/query/process_tracker.rs`) files in it.

We need to move it to `scheduler` crate and name them `types.rs` and `query.rs` accordingly. Re-export them just so that we can use them in `diesel_models` crate.  

This helps keep all the process_tracker related files in a single place.

### PR DESCRIPTION
**Pull request by pixincreate**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Re-export `DieselArray` from `diesel_models` crate so that it could be used in `scheduler` crate (used by `scheduler/src/db/types.rs`)
- Import `diesel` and `router_derive` crate in `scheduler` crate as both the files (`types.rs` and `query.rs`) depend on it
- Re-export `types.rs` from `scheduler` crate just so that it can be used by `diesel_models` crate

TL;DR:

- This will come in useful to keep all the `process_tracker` related files in same place
- No code changes has been made here but only the movement of files and corresponding fixes

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

The motivation here is that we need to keep sandbox similar feature / code. Since `process_tracker` related files were in `diesel_models`, and all other `process_tracker` related files are kept in `scheduler` crate, it is worth moving them.

Resolves #4921

## How did you test it?

Below table shows the `name`s of the `process_tracker` and these can be tested (Even if one works, it is enough given that the change introduced should affect all 7 names the same way). 

|process_tracker_names|
|-|
|EXECUTE_REFUND|
|STRPE_ATTACH_EXTERNAL_ACCOUNT|
|SYNC_REFUND|
|OUTGOING_WEBHOOK_RETRY|
|API_KEY_EXPIRY|
|DELETE_TOKENIZE_DATA_WORKFLOW|
|PAYMENTS_SYNC|

I did some payments by running locally and checked whether processes are being scheduled:

> [!NOTE]
> Disable webhooks (if any)

- Payment Sync workflow (checked with `checkout`)
	```curl
	curl --location 'http://Localhost:8080/payments/pay_bZaBO0FHxzBCP3X9VBiR' \
	--header 'Accept: application/json' \
	--header 'api-key: dev_z7Z1SkbLl7rma0cPHfVQBsivxayVgegNqS8qs7LYRd0D8zgSz2cUIbRThyTVXz5R'
	```

	image
	
	image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Comment by pixincreate** [comment]
closing this PR in favor of https://github.com/juspay/hyperswitch/pull/4849
